src/containers/ConfirmPassword/confirmPass.jsx

```diff
@@ -33,85 +34,71 @@ class ConfirmPass extends Component {
-                ],
-              })(
-                <Input.Password
-                  prefix={
-                    <Icon type="lock" className="confirm-pass__form__icon" />
-                  }
-                  placeholder="Enter your password"
-                  onChange={() => this.setState({ errorMesage: '' })}
-                  className="confirm-pass__input"
-                />
-              )}
-            </Form.Item>
-
-            {errorMesage && <p style={{ color: 'red' }}>{errorMesage}</p>}
-
-            <Button type="primary" size="large" onClick={this.handleSubmit}>
-              Proceed To Edit
-            </Button>
-          </Form>
-          <ConfirmImg className="confirm-pass__img" />
-        </section>
-      </div>
-    );
-  }
-}
+  return (
+    <div className="confirm-pass">
+      <Header text="Confirm Password" handleBack={handelGoBack} />
+
+      <section className="confirm-pass__body">
+        <p className="confirm-pass__text">
+          This is to make sure it&apos;s you!
+        </p>
+        <Form className="confirm-pass__form" onSubmit={handleSubmit}>
+          <Form.Item>
+            {getFieldDecorator('password', {
+              rules: [
+                {
+                  required: true,
+                  message: 'Enter Your Password',
+                },
+              ],
+            })(
+              <Input.Password
+                prefix={
+                  <Icon type="lock" className="confirm-pass__form__icon" />
+                }
+                placeholder="Enter your password"
+                onChange={handleChange}
+                className="confirm-pass__input"
+              />
+            )}
+          </Form.Item>
+
+          {errorMessage && <p style={{ color: 'red' }}>{errorMessage}</p>}
+
+          <Button type="primary" size="large" onClick={handleSubmit}>
+            Proceed To Edit
+          </Button>
+        </Form>
+        <ConfirmImg className="confirm-pass__img" />
+      </section>
+    </div>
+  );
+};
 
 const confirmPasswordForm = Form.create({ name: 'confirm pass' })(ConfirmPass);
 
 ConfirmPass.propTypes = {
-  history: PropTypes.shape({
-    goBack: PropTypes.func.isRequired,
-    push: PropTypes.func.isRequired,
-  }).isRequired,
+  handelGoBack: PropTypes.func.isRequired,
+  handleChange: PropTypes.func.isRequired,
+  handleErrorMessage: PropTypes.func.isRequired,
+  handlePush: PropTypes.func.isRequired,
   form: PropTypes.shape({
     validateFields: PropTypes.func.isRequired,
     getFieldDecorator: PropTypes.func.isRequired,
   }).isRequired,
-
+  errorMessage: PropTypes.string.isRequired,
   firebase: PropTypes.shape({
     doSignInWithEmailAndPassword: PropTypes.func.isRequired,
     auth: PropTypes.object.isRequired,
   }).isRequired,
 };
 
-const AuthConfirmaPassword = compose(
-  withAuth,
-  withFirebase
-)(confirmPasswordForm);
-
-export default AuthConfirmaPassword;
+export default withFirebase(confirmPasswordForm);
```

Comment:
You are using firebase in the index file which is enough so if you want anything fro firebase to use here you can pass it via the props but I think if you move that function you will not even need withFirebase here